src/Analysers/AttributeAnnotationFactory.php

```diff
@@ -80,6 +80,9 @@ public function build(\Reflector $reflector, Context $context): array
                                 $instance->nullable = $nullable ?: Generator::UNDEFINED;
 
                                 if ($rp->isPromoted()) {
+                                    // ensure each property has its own context
+                                    $instance->_context = clone $instance->_context;
```

Comment:
Not so keen on `clone` - could we use this instead?
```suggestion
                                    $instance->_context = new Context([], $instance->_context);
```

Reply:
Fixed :-)